routing/pathfind.go

```diff
@@ -743,6 +739,10 @@ func findPath(g *graphParams, r *RestrictParams, source, target Vertex,
 			processEdge(reverseEdge.sourceNode, reverseEdge.edge,
 				bandWidth, pivot)
 		}
+
+		// We have visited the node and explored all incoming edges so
+		// we add it to our visited map.
+		visitedNodes[pivot] = struct{}{}
```

Comment:
This could be moved here in the commit where `visitedNodes` is introduced.

Reply:
Fixed